### ISSUE TITLE
Support (loose) string-to-property-map matching in `icu_properties`

### ISSUE THREAD

**Issue by skius** [unicode-org/icu4x#3559]
`icu_properties` does not support obtaining a CodePointMapData (or CodePointSet) from the string of a property name for many properties. It's supported for ECMA-262 binary properties through `load_for_ecma_262_unstable`, but, e.g., [`load_word_break`](https://docs.rs/icu_properties/latest/icu_properties/maps/fn.load_word_break.html) cannot be accessed given a string; the client would have to perform property-name-to-`icu_properties`-functions mapping themselves. Loose matching should also be supported.

UnicodeSet could also support more properties with those features.

**Comment by sffc** [comment]
Discussion: revisit when we have a clearer picture of the set of properties needed by the transliterator.

**Comment by Manishearth** [comment]
Yep, when I added the ecma402 function my hope was to potentially add this one as well but we decided to wait for a concrete use case. It does seem like we have one now!

**Comment by sffc** [comment]
@makotokato added some additional properties needed for Segmenter data in #4175
